cmd/gcp-controller-manager/app/gke_signer.go

```diff
@@ -83,57 +69,62 @@ func newGKESigner(kubeConfigFile string, recorder record.EventRecorder, client c
 }
 
 func (s *gkeSigner) handle(csr *capi.CertificateSigningRequest) error {
-	start := time.Now()
+	metricRecord := csrmetrics.CSRSigningStartRecorder()
 	if !certificates.IsCertificateRequestApproved(csr) {
 		return nil
 	}
 	klog.Infof("gkeSigner triggered for %q", csr.Name)
 	csr, err := s.sign(csr)
 	if err != nil {
-		csrSigningStatus.WithLabelValues("sign_error").Inc()
-		csrSigningLatency.WithLabelValues("sign_error").Observe(time.Since(start).Seconds())
+		metricRecord(csrmetrics.CSRSigningStatusSignError)
 		return fmt.Errorf("error auto signing csr: %v", err)
 	}
 	_, err = s.client.CertificatesV1beta1().CertificateSigningRequests().UpdateStatus(csr)
 	if err != nil {
-		csrSigningStatus.WithLabelValues("update_error").Inc()
-		csrSigningLatency.WithLabelValues("update_error").Observe(time.Since(start).Seconds())
+		metricRecord(csrmetrics.CSRSigningStatusUpdateError)
 		return fmt.Errorf("error updating signature for csr: %v", err)
 	}
 	klog.Infof("CSR %q signed", csr.Name)
-	csrSigningStatus.WithLabelValues("signed").Inc()
-	csrSigningLatency.WithLabelValues("signed").Observe(time.Since(start).Seconds())
+	metricRecord(csrmetrics.CSRSigningStatusSigned)
 	return nil
 }
 
 // Sign will make an external call to GKE order to sign the given
 // *capi.CertificateSigningRequest, using the gkeSigner's
 // kubeConfigFile.
 func (s *gkeSigner) sign(csr *capi.CertificateSigningRequest) (*capi.CertificateSigningRequest, error) {
+	metricRecord := csrmetrics.OutboundRPCStartRecorder("container.webhook.Sign")
```

Comment:
Move this into the WithExponentialBackoff below and record each call to s.webhook.RestClient.Post().Body(csr).Do()

Reply:
Also, check result.StatusCode in this backoff callback, to retry on non-RPC errors

Reply:
Done. I believe.